### ISSUE TITLE
The tests aren't cleaning up after themselves

### ISSUE THREAD

**Issue by Skarlso** [gaia-pipeline/gaia#10]
When I run `go test` a bunch of things are left under `pipeline` that pollute the work tree.

```bash
~/golang/src/github.com/gaia-pipeline/gaia makefile_refactor* 10s
❯ git status
On branch makefile_refactor
Your branch is up-to-date with 'skarlso/makefile_refactor'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	pipeline/.circleci/
	pipeline/.gitignore
	pipeline/.goreleaser.yml
	pipeline/Gopkg.lock
	pipeline/Gopkg.toml
	pipeline/LICENSE
	pipeline/Makefile
	pipeline/README.rst
	pipeline/cmd/
	pipeline/docker/
	pipeline/frontend/
	pipeline/gaia.go
	pipeline/handlers/
	pipeline/pipeline/
	pipeline/plugin/
	pipeline/scheduler/
	pipeline/store/
	pipeline/vendor/
```

**Comment by michelvocks** [comment]
I'm actually a bit ashamed of the tests. 😣
They are really messed up and most times they are not working.

**Comment by Skarlso** [comment]
Don't worry about it. :) I'll clean them up as I go through them. ;)

**Comment by Skarlso** [comment]
Fixed by #12
